crates/cli/src/utils/cmd.rs

```diff
@@ -447,22 +449,33 @@ pub async fn print_traces(
     result: &mut TraceResult,
     decoder: &CallTraceDecoder,
     verbose: bool,
+    json: bool,
+    state_changes: bool,
 ) -> Result<()> {
     let traces = result.traces.as_mut().expect("No traces found");
 
-    println!("Traces:");
+    if !json {
+        println!("Traces:");
+    }
+
     for (_, arena) in traces {
         decode_trace_arena(arena, decoder).await?;
-        println!("{}", render_trace_arena_with_bytecodes(arena, verbose));
+        println!("{}", render_trace_arena_inner(arena, verbose, json, state_changes));
     }
-    println!();
 
-    if result.success {
-        println!("{}", "Transaction successfully executed.".green());
-    } else {
-        println!("{}", "Transaction failed.".red());
+    if json {
+        return Ok(());
+    }
+
+    if !json {
+        println!();
+        if result.success {
+            println!("{}", "Transaction successfully executed.".green());
+        } else {
+            println!("{}", "Transaction failed.".red());
+        }
+        println!("Gas used: {}", result.gas_used);
```

Comment:
This second check for !json isn't necessary due to the early return above

Please use the new `sh_println!`, this will make it compatible with the --quiet flag

Reply:
Updated, thanks!